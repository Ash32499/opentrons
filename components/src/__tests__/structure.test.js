// structure components tests
import React from 'react'
import Renderer from 'react-test-renderer'

import {TitleBar} from '..'

describe('TitleBar', () => {
  test('adds an h1 with the title', () => {
    const heading = Renderer.create(
      <TitleBar title='hello' />
    ).root.findByType('h1')

    expect(heading).toBeDefined()
    expect(heading.children).toEqual(['hello'])
  })

  test('adds an optional h2 with the subtitle', () => {
    const heading = Renderer.create(
      <TitleBar title='hello' subtitle='world' />
    ).root.findByType('h2')

    expect(heading).toBeDefined()
    expect(heading.children).toEqual(['world'])
  })

  test('renders TitleBar without subtitle correctly', () => {
    const tree = Renderer.create(
      <TitleBar title='foo' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders TitleBar with subtitle correctly', () => {
    const tree = Renderer.create(
      <TitleBar title='foo' subtitle='bar' />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
