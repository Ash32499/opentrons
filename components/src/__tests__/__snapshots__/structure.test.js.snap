// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders Card correctly 1`] = `
<section
  className="card"
>
  <h3
    className="card_title"
  >
    title
  </h3>
  <div
    className="card_content"
  >
    children children children
  </div>
</section>
`;

exports[`LabeledValue renders LabeledValue correctly 1`] = `
<div
  className="labeled_value"
>
  <p
    className="labeled_value_label"
  >
    Label
    :
  </p>
  <p
    className="labeled_value_value"
  >
    Value
  </p>
</div>
`;

exports[`PageTabs renders PageTabs correctly 1`] = `
<nav
  className="page_tabs"
>
  <a
    className="tab_link active_tab_link"
    href="/page1"
    onClick={[Function]}
  >
    <h3
      className="tab_title"
    >
      Page1
    </h3>
  </a>
  <span
    className="tab_link"
    to="/page2"
  >
    <h3
      className="tab_title"
    >
      Page2
    </h3>
  </span>
</nav>
`;

exports[`RefreshCard renders correctly 1`] = `
<section
  className="card"
>
  <h3
    className="card_title"
  />
  <div
    className="card_content"
  >
    <button
      className="button_flat button_icon refresh_card_icon"
      disabled={true}
      onClick={undefined}
      title={undefined}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="button_only_icon spin"
        fill="currentColor"
        height={undefined}
        style={undefined}
        version="1.1"
        viewBox="0 0 512 512"
        width={undefined}
        x={undefined}
        y={undefined}
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fillRule="evenodd"
        />
      </svg>
    </button>
    child1 child2 child3
  </div>
</section>
`;

exports[`TitleBar renders TitleBar with back button correctly 1`] = `
<header
  className="title_bar"
>
  <button
    className="button_flat back_button"
    disabled={undefined}
    onClick={[Function]}
    title="back"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="icon"
      fill="currentColor"
      height={undefined}
      style={undefined}
      version="1.1"
      viewBox="0 0 24 24"
      width={undefined}
      x={undefined}
      y={undefined}
    >
      <path
        d="M15.429 17.019l-4.242-4.59 4.242-4.59-1.312-1.41-5.545 6 5.545 6z"
        fillRule="evenodd"
      />
    </svg>
    Back
  </button>
  <div
    className="title_wrapper"
  >
    <h1
      className="title"
    >
      foo
    </h1>
    <span
      className="separator"
    >
      |
    </span>
    <h2
      className="subtitle"
    >
      bar
    </h2>
  </div>
</header>
`;

exports[`TitleBar renders TitleBar with subtitle correctly 1`] = `
<header
  className="title_bar"
>
  <div
    className="title_wrapper"
  >
    <h1
      className="title"
    >
      foo
    </h1>
    <span
      className="separator"
    >
      |
    </span>
    <h2
      className="subtitle"
    >
      bar
    </h2>
  </div>
</header>
`;

exports[`TitleBar renders TitleBar without subtitle correctly 1`] = `
<header
  className="title_bar"
>
  <div
    className="title_wrapper"
  >
    <h1
      className="title"
    >
      foo
    </h1>
  </div>
</header>
`;
